lib/ansible/playbook/conditional.py

```diff
@@ -92,11 +92,13 @@ def evaluate_conditional(self, templar, all_vars):
             ds = getattr(self, '_ds')
 
         try:
-            # this allows for direct boolean assignments to conditionals "when: False"
-            if isinstance(self.when, bool):
-                return self.when
-
             for conditional in self.when:
+                # shortcut for direct boolean assignments to conditionals: "when: False"
+                if isinstance(conditional, bool):
```

Comment:
I would put this in _check_conditional, that way this block can be reduced to an `all(self._check_conditional(x, templar, all_vars) for x in self.when)`. This also means that all checks are done in the function that does the checks, instead of doing it in two places.